iofog-agent-daemon/src/main/java/org/eclipse/iofog/process_manager/DockerUtil.java

```diff
@@ -311,21 +327,33 @@ private long getStartedTime(String startedTime) {
      */
     public MicroserviceStatus getMicroserviceStatus(String containerId, String microServiceUuid) {
     	LoggingService.logDebug(MODULE_NAME , "get microservice status");
-        InspectContainerResponse inspectInfo = dockerClient.inspectContainerCmd(containerId).exec();
-        ContainerState containerState = inspectInfo.getState();
+        InspectContainerResponse inspectInfo;
         MicroserviceStatus result = new MicroserviceStatus();
-        if (containerState != null) {
-            if (containerState.getStartedAt() != null) {
-                result.setStartTime(getStartedTime(containerState.getStartedAt()));
-            }
-
-            MicroserviceState microserviceState = containerToMicroserviceState(containerState);
-             result.setStatus(MicroserviceState.EXITING.equals(microserviceState) && RestartStuckChecker.isStuck(microServiceUuid)
-                ? MicroserviceState.STUCK_IN_RESTART
-                : microserviceState);
+        try {
+            inspectInfo = dockerClient.inspectContainerCmd(containerId).exec();
+            ContainerState containerState = inspectInfo.getState();
+            if (containerState != null) {
+                if (containerState.getStartedAt() != null) {
+                    result.setStartTime(getStartedTime(containerState.getStartedAt()));
+                }
 
-            result.setContainerId(containerId);
-            result.setUsage(containerId);
+                MicroserviceState microserviceState = containerToMicroserviceState(containerState);
+                result.setStatus(MicroserviceState.EXITING.equals(microserviceState) && RestartStuckChecker.isStuck(microServiceUuid)
```

Comment:
Rewrite this ternary logic! Make another function that maps the boolean inputs to and output if it makes it tidier